std/http/_io.ts

```diff
@@ -171,21 +171,21 @@ function parseTrailer(field: string | null): Headers | undefined {
 }
 
 export async function writeChunkedBody(
-  w: Deno.Writer,
+  w: BufWriter,
```

Comment:
```suggestion
  w: Deno.Writer & { flush(): Promise<void> }
```

This would allow other "flushable" writers without specifically requiring a BufWriter. Maybe even creat and export a `Flushable` interface and export it and then require `Deno.Writable & Flushable`.

Reply:
Hmm, is this really an improvement? I think the previous interface being needlessly unspecific created the situation where a `BufWriter` was wrapped in a `BufWriter` unnecessarily.

I also feel like both `Deno.Writer & { flush(): Promise<void> }` and `Deno.Writable & Flushable` are harder to read and understand than `BufWriter`.

Reply:
But it isn't always a BufWriter, BufWriter is an implementation of a concept and may not always be what is used downtream. The previous interface wasn't causing the problem, the code was not flushing the data, irrespective of the interface. There is a reason why we use `Deno.Writer` instead of a specific  implementation.

Reply:
I know the `writeChunkedBody` function is exported, but I expected it to be exported only for testing purposes (though it doesn't have any tests at the moment).

It's only called in one place in the same file. To me, it looks more like a function that was created to make `writeResponse` a bit shorter and easier to digest.

There's nothing technically wrong with using `Deno.Writer & { flush(): Promise<void> }` rather than `BufWriter`. I just think that that interface is a bit overengineered for a function that could be inlined.

Reply:
Yeah, ok, fair enough, in that it is exported, but from a "private" module, so it is fine coupling to an implementation interface.